graylog2-server/src/main/java/org/graylog2/inputs/transports/KafkaNewTransport.java

```diff
@@ -0,0 +1,491 @@
+    }
+
+    @Override
+    public void setMessageAggregator(CodecAggregator ignored) {
+    }
+
+    @Override
+    public void doLaunch(final MessageInput input) throws MisfireException {
+        serverStatus.awaitRunning(new Runnable() {
+            @Override
+            public void run() {
+                lifecycleStateChange(Lifecycle.RUNNING);
+            }
+        });
+
+        // listen for lifecycle changes
+        serverEventBus.register(this);
+
+        final Properties props = new Properties();
+
+        props.put("group.id", GROUP_ID);
+        props.put("client.id", "gl2-" + nodeId + "-" + input.getId());
+        props.put(ConsumerConfig.FETCH_MIN_BYTES_CONFIG, String.valueOf(configuration.getInt(CK_FETCH_MIN_BYTES)));
+        props.put(ConsumerConfig.FETCH_MAX_WAIT_MS_CONFIG, String.valueOf(configuration.getInt(CK_FETCH_WAIT_MAX)));
+        props.put("auto.offset.reset", configuration.getString(CK_OFFSET_RESET, DEFAULT_OFFSET_RESET));
+        // Default auto commit interval is 60 seconds. Reduce to 1 second to minimize message duplication
+        // if something breaks.
+        props.put("commit.interval.ms", "1000");
+        // Set a consumer timeout to avoid blocking on the consumer iterator.
+        props.put("consumer.timeout.ms", "1000");
+        props.put("bootstrap.servers", configuration.getString(CK_BOOTSTRAP));
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,ByteArrayDeserializer.class.getName());
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,ByteArrayDeserializer.class.getName());
+
+        // SSL seettings
+
+        if(configuration.getString(CK_SSL).equals("true")) {
+            props.put("ssl.keystore.location", configuration.getString(CK_SSL_KEYSTORE_LOCATION));
+            props.put("ssl.keystore.password", configuration.getString(CK_SSL_KEYSTORE_PASSWORD));
+            props.put("ssl.key.password", configuration.getString(CK_SSL_KEY_PASSWORD));
+            props.put("ssl.truststore.location", configuration.getString(CK_SSL_TRUSTSTORE_LOCATION));
+            props.put("ssl.truststore.password", configuration.getString(CK_SSL_TRUSTSTORE_PASSWORD));
+            props.put("ssl.enabled.protocols", configuration.getString(CK_SSL_ENABLED_PROTOCOL));
+            props.put("security.protocol", "SSL");
+        }
+
+        final int numThreads = configuration.getInt(CK_THREADS);
+        consumer = new KafkaConsumer<>(props);
+        String zkConnect = configuration.getString(CK_ZOOKEEPER);
+
+        ZkClient zkClient = new ZkClient(zkConnect, 30000, 30000, ZKStringSerializer$.MODULE$);
+        ZkUtils zkUtils = ZkUtils.apply(zkClient, false);
+
+        List<String> subscribedTopics = new ArrayList<String>();
+
+        List<String> topics = JavaConversions.seqAsJavaList(zkUtils.getAllTopics());
+
+        topics.stream()
+            .forEach(topicName->{
+                if(topicName!=null && topicName.matches(configuration.getString(CK_TOPIC_FILTER))) {
```

Comment:
This is duplicating functionality in KafkaClient, which already supports pattern matching with the KafkaClient.subsribe(Pattern) API call, and iterating through all topics like this is potentially really inefficient (and doesn't handle updates to available topics.

It'd be better to avoid requiring Zookeeper as a parameter at all and just let the KafkaClient library handle this.

Reply:
Sent in a pull request to address this.

Reply:
I will take a look